lib/facility.rb

```diff
@@ -1,14 +1,77 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone,
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_specs)
+    @name = facility_specs[:name]
+    @address = facility_specs[:address]
+    @phone = facility_specs[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
-end
+
+  def register_vehicle(vehicle)
```

Comment:
Consider making collect fees its own separate helper function. Promote SRP by using many small functions that do one thing really well.